data_access.go

```diff
@@ -156,7 +160,24 @@ func (s *DataAccessSyncer) importDataAccess(config *data_access.DataAccessSyncCo
 			}
 		}
 
-		err = fileCreator.AddAccessProvider([]dap.AccessProvider{da})
+		// copy AccessObjects from this role to all roles that have a GRANT on this one
+		for _, grantee := range grantOfEntities {
+			if grantee.GrantedTo == "ROLE" {
+				if _, f := accessProviderMap[grantee.GranteeName]; !f {
+					accessProviderMap[grantee.GranteeName] = &dap.AccessProvider{
+						ExternalId: grantee.GranteeName,
+						Name:       grantee.GranteeName,
+					}
+				}
+				granteeDa := accessProviderMap[grantee.GranteeName]
+				granteeDa.AccessObjects = append(granteeDa.AccessObjects, da.AccessObjects...)
```

Comment:
Don't you have a problem with the order in which roles are handled?
If grantedTo role is not processed yet, the AccessObjects for it would be empty still (and only filled in one of the following loops). Or am I missing something?

Reply:
No because it is the other way around actually. The role that you are visiting in the loop is sharing its objects to the grantee not inheriting. That is why I check wether the map contains the grantee already and if not create a placeholder and already fill the access objects. When we visit that role then we just add it’s own objects to it